ci: add initial appveyor support

jniles opened the pull request:
Start running a few tests using appveyor to take the pressure of TravisCI.  The end goal is to run our tests in two CI environments to speed up the tests and also give us better coverage.  For example, right now we only test on Travis on trusty.  It would be nice to move over to Bionic or even Windows support as referenced in https://github.com/IMA-WorldHealth/bhima/issues/2477.

Closes https://github.com/IMA-WorldHealth/bhima/issues/887.

jniles committed:
ci: add initial appveyor support

Start running a few tests using appveyor.

jniles committed:
test: add appveyor to bors

jniles committed:
test: remove server-unit from TravisCI

This commit removes the server-unit tests from Travis CI testing.  It is
handled instead by Appveyor.

Closes #887.

mbayopanda commented on test/server-unit/pdf.spec.js:
```diff
@@ -53,7 +53,7 @@ function PDFRenderUnitTest() {
   });
 
-  it('#pdf.render() renders an identical PDF given an HTML template', async () => {
+  it.skip('#pdf.render() renders an identical PDF given an HTML template', async () => {
```
I think there is a good reason for this `.skip`, i would like to know it was running on Travis why you skipped it here.

jniles replied:
I skipped this because I didn't want to set up XVFB.  Let me see if I can reproduce it with XVFB like on Travis CI.

jniles replied:
Well, it looks like the PDF output isn't deterministic: https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4437.  I'm going to just remove this test since it won't ever work properly.

mbayopanda commented on sh/test.sh:
```diff
@@ -27,9 +27,9 @@ startfold "Running Integration Tests..." "test-integration";
-endfold "server-unit" ;
+# startfold "Running server Unit Tests ......" "server-unit"
+# ./node_modules/.bin/mocha --recursive --exit test/server-unit
+# endfold "server-unit" ;
```
Is it possible to add a comment about the fact that server-unit tests are running on `appveyor` or just remove these commented codes.

jniles replied:
Good point.  I'll remove this code.

mbayopanda reviewed (approved):
LGTM 👍

jniles committed:
chore: refactor testing code

Reshuffles the testing code to consolidate all CI-related setup into a
single setup file.  Removes the PDF comparison test since the output of
wkhtmltopdf is not deterministic [1].

[1] https://github.com/wkhtmltopdf/wkhtmltopdf/issues/4437

jniles committed:
test: run client unit tests on AppVeyor

Moves the client-unit tests to AppVeyor.
